baseplate/frameworks/queue_consumer/kafka.py

```diff
@@ -19,6 +19,7 @@
 from prometheus_client import Counter
 from prometheus_client import Gauge
 from prometheus_client import Histogram
+from typing_extensions import Self
```

Comment:
Do you think it's worth doing

```python
try:
    from typing import Self  # Python >= 3.11
except ImportError:
    from typing_extensions import Self
```

and then doing this in `pyproject.toml`:

```toml
typing-extensions = {version = "^4.11.0", python = "<3.11"}
```

?